src/material/tree/padding.ts

```diff
@@ -5,21 +5,51 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
+import {coerceNumberProperty} from '@angular/cdk/coercion';
 import {CdkTreeNodePadding} from '@angular/cdk/tree';
 import {Directive, Input} from '@angular/core';
 
+/** Regex used to split a string on its CSS units. */
+const cssUnitPattern = /([A-Za-z%]+)$/;
+
 /**
  * Wrapper for the CdkTree padding with Material design styles.
  */
 @Directive({
   selector: '[matTreeNodePadding]',
   providers: [{provide: CdkTreeNodePadding, useExisting: MatTreeNodePadding}]
 })
+// tslint:disable-next-line: coercion-types
 export class MatTreeNodePadding<T> extends CdkTreeNodePadding<T> {
 
   /** The level of depth of the tree node. The padding will be `level * indent` pixels. */
-  @Input('matTreeNodePadding') level: number;
+  @Input('matTreeNodePadding')
+  get level(): number { return this._level; }
+  set level(value: number) {
+    // Set to null as the fallback value so that _setPadding can fall back to the node level if the
+    // consumer set the directive as `cdkTreeNodePadding=""`. We still want to take this value if
+    // they set 0 explicitly.
+    this._level = coerceNumberProperty(value, null)!;
+    this._setPadding();
+  }
+  _level: number;
 
   /** The indent for each level. Default number 40px from material design menu sub-menu spec. */
-  @Input('matTreeNodePaddingIndent') indent: number;
```

Comment:
I'm wondering if we should add comments for this new pattern. The reason why we need this (for other reviewers not being aware of this): TS v4 does not allow us to override an inherited accessor with a non-accessor class member. The re-declared input purely existed to tell the Angular compiler to rename the inherited input from `cdkTreeNode<..>` to `matTreeNode`.

We might get around it better (without having to duplicate the full getters/setters) by renaming the inputs in the `@Directive` metadata. I think it could work here as this class seems to be an actual directive (not an abstract one)

Reply:
I’ll try that.

Reply:
Seems like it breaks VE 

https://github.com/angular/components/pull/20368#discussion_r474127683